converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/FhirTransactionPipeline.cs

```diff
@@ -23,100 +33,161 @@ public class FhirTransactionPipeline : IFhirTransactionPipeline
     private readonly IFhirTransactionExecutor _fhirTransactionExecutor;
 
     private readonly IReadOnlyList<FhirTransactionRequestResponsePropertyAccessor> _requestResponsePropertyAccessors;
+    private readonly IAsyncPolicy _retryPolicy;
+    private readonly IExceptionStore _exceptionStore;
+    private readonly AsyncTimeoutPolicy _timeoutPolicy;
+    private readonly ILogger<FhirTransactionPipeline> _logger;
 
     public FhirTransactionPipeline(
         IEnumerable<IFhirTransactionPipelineStep> fhirTransactionPipelines,
         IFhirTransactionRequestResponsePropertyAccessors fhirTransactionRequestResponsePropertyAccessors,
-        IFhirTransactionExecutor fhirTransactionExecutor)
+        IFhirTransactionExecutor fhirTransactionExecutor,
+        IExceptionStore exceptionStore,
+        IOptions<RetryConfiguration> retryConfiguration,
+        ILogger<FhirTransactionPipeline> logger)
     {
-        EnsureArg.IsNotNull(fhirTransactionPipelines, nameof(fhirTransactionPipelines));
-        EnsureArg.IsNotNull(fhirTransactionRequestResponsePropertyAccessors, nameof(fhirTransactionRequestResponsePropertyAccessors));
-        EnsureArg.IsNotNull(fhirTransactionExecutor, nameof(fhirTransactionExecutor));
-
-        _fhirTransactionPipelines = fhirTransactionPipelines;
-        _fhirTransactionExecutor = fhirTransactionExecutor;
-
-        _requestResponsePropertyAccessors = fhirTransactionRequestResponsePropertyAccessors.PropertyAccessors;
+        _fhirTransactionPipelines = EnsureArg.IsNotNull(fhirTransactionPipelines, nameof(fhirTransactionPipelines));
+        _requestResponsePropertyAccessors = EnsureArg.IsNotNull(fhirTransactionRequestResponsePropertyAccessors?.PropertyAccessors, nameof(fhirTransactionRequestResponsePropertyAccessors));
+        _fhirTransactionExecutor = EnsureArg.IsNotNull(fhirTransactionExecutor, nameof(fhirTransactionExecutor));
+        _exceptionStore = EnsureArg.IsNotNull(exceptionStore, nameof(exceptionStore));
+        EnsureArg.IsNotNull(retryConfiguration?.Value, nameof(retryConfiguration));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _timeoutPolicy = Policy.TimeoutAsync(retryConfiguration.Value.TotalRetryDuration);
+        _retryPolicy = Policy
+            .Handle<RetryableException>()
+            .WaitAndRetryForeverAsync(
+                (retryAttempt, exception, context) =>
+                {
+                    return TimeSpan.FromSeconds(Math.Min(60, Math.Pow(2, retryAttempt)));
+                },
+                (exception, retryCount, timeSpan, context) =>
+                {
+                    var changeFeedEntry = (ChangeFeedEntry)context[nameof(ChangeFeedEntry)];
```

Comment:
I'd say either put everything into the context or nothing. Presumably we want to use the context to avoid creating a closure, so I think we can shove `_exceptionStore` in there too. Visual Studio (and maybe Rider) allows you to see what variables are captured by hovering over the `=>` for an anonymous function